Small crate animation adjustments and coffin animation

MNarath1 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This PR fixes a small oversight of mine in the crate animation leading it to loop at the end during open cause i went over 180 degree in the coordinate system.
Fixes some icon's in the dmi file that had the wrong Position i do not know how this happened but it made the crate animation look really weird for the effected crates like the botany crates.
Improves the freezer crate sprite making the animation look better overall.
Adds the needed sprites for coffin animation and allow coffins to be animated.
Uses icon_door_override to remove unneeded sprites wherever possible.
https://streamable.com/46tlj6

## Why It's Good For The Game

Small improvements to the animation overall and adding coffin animation

## Changelog
:cl:
add: coffin animation
tweak: angle of crate animation
tweak: freezer sprite and animation parameter
fix: some crate sprites breaking animation
tweak: use icon_door_override to allow for removal of unnecessary sprites
tweak: changing crate sprite names for consistancy
del: unnecessary sprites after using icon_door_override whenever possible
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

MNarath1 committed:
small animation changes including back sprite coffin

coffin back sprite is still unrefined and the transition between the back and the open sprites needs to be better

MNarath1 committed:
changes to sprite name for consistency and other stuff

*changes a lot of sprite names to get a sort of consistancy
*makes the solar crate a subtype of the electrical crate instead of normal crates same with rcd crate but instead a subtype of the normal engineering crate
*use icon_door_override and icon_door to override the correct icons to allow the deletion of the obsolete sprites and reduce the crates.dmi file size wherever possible

MNarath1 committed:
properly implements coffin animation

*tweaks animation values for coffin animation to make it look good
*replaces the back sprite for coffins with a better one

MNarath1 committed:
partially reverts some of my obj path changes

yes those Consistencies are cringe but if i change those its just going to get more confusing

MNarath1 committed:
fixes small icon position missmatch breaking animation

MNarath1 committed:
small changes to the freezer crate sprite and animation parameter

park66665 commented on code/modules/mining/abandoned_crates.dm:
```diff
@@ -172,7 +171,7 @@
 				cut_overlays()
 				add_overlay("securecrateg")
-				add_overlay("[icon_state]_door") //needs to put the door overlayer back cause of this snowflake code
+				add_overlay("icon_door ? [icon_door]_door : [icon_state]_door") //needs to put the door overlayer back cause of this snowflake code
```
```suggestion
				add_overlay("[icon_door || icon_state]_door") //needs to put the door overlayer back cause of this snowflake code
```

park66665 reviewed (commented):
The PR is generally okay.

MNarath1 committed:
Implementing suggested change

Using an or operator instead of checking for icon_door is a pretty good idea honestly causes looks like once the first part of OR is already true it does not check for the second part. co-authored-by: park66665 <park66665@user.noreply.github.com>

park66665 reviewed (approved):
All clear. This PR is good to go.
